vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/BatchRegistrationDialog.java

```diff
@@ -99,60 +94,83 @@ private class RegisterBatchDialogHandler implements Serializable {
 
     @Serial
     private static final long serialVersionUID = -2692766151162405263L;
-    private final List<ComponentEventListener<SampleRegistrationEvent>> listeners = new ArrayList<>();
-    private final List<ComponentEventListener<UserCancelEvent<SampleRegistrationDialog>>> cancelListeners = new ArrayList<>();
+    private final List<ComponentEventListener<SampleRegistrationEvent>> sampleRegistrationListeners = new ArrayList<>();
+    private final List<ComponentEventListener<BatchRegistrationEvent>> batchRegistrationListeners = new ArrayList<>();
+    private final List<ComponentEventListener<UserCancelEvent<BatchRegistrationDialog>>> cancelListeners = new ArrayList<>();
 
     public RegisterBatchDialogHandler() {
+      setNavigationListeners();
+      setSubmissionListeners();
       resetDialogueUponClosure();
-      setBatchInformationButtonsListeners();
-      setSampleRegistrationSubmission();
-      setTabSelectionListener();
     }
 
-    private void setBatchInformationButtonsListeners() {
+    private void setNavigationListeners() {
       batchInformationLayout.nextButton.addClickListener(
           event -> tabStepper.setSelectedTab(sampleInformationTab));
+      setTabSelectionListener();
+    }
+
+    private void setTabSelectionListener() {
+      tabStepper.addSelectedChangeListener(event -> {
```

Comment:
what is a `tabStepper` ?

Reply:
A [stepper](https://mdbootstrap.com/docs/standard/components/stepper/) is a frontend component which allows to switch the content shown dependent on which element within the stepper the user clicks. Since vaadin doesn'T offer this specifically we use the [tabsheet ](https://vaadin.com/docs/latest/components/tabs)component to implement this functionality.

Reply:
a nice one, learned sth today :D